__tests__/app.test.js

```diff
@@ -280,3 +280,69 @@ describe("POST /api/articles/:article_id/comments", () => {
       });
   });
 });
+describe("PATCH /api/articles/:article_id", () => {
+  it("200: updates the article and responds with the updated article", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inc_votes: 1 })
+      .expect(200)
+      .then(({ body: { article } }) => {
+        expect(article).toMatchObject({
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          votes: 101,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });
+        expect(typeof article.created_at).toBe("string");
+      });
+  });
+  it("200: decreases the number of votes if passed a negative value", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inc_votes: -200 })
+      .expect(200)
+      .then(({ body: { article } }) => {
+        expect(article.votes).toBe(-100);
+      });
+  });
+  it("400: responds with the correct error message when the article id is invalid", () => {
+    return request(app)
+      .patch("/api/articles/not_an_id")
+      .send({ inc_votes: 1 })
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  it("404: responds with the correct error message when the article id is valid but non-existent", () => {
+    return request(app)
+      .patch("/api/articles/999")
+      .send({ inc_votes: 1 })
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("No Article With That Id Found");
+      });
+  });
+  it("400: responds with correct error messgae when passed information in the wrong format", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inv_votes: 1 })
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Incorrect Information For Request");
+      })
+      .then(() => {
+        return request(app)
+          .patch("/api/articles/1")
+          .send({ inc_votes: "hello" })
+          .expect(400)
+          .then(({ body: { msg } }) => {
+            expect(msg).toBe("Incorrect Information For Request");
+          });
+      });
+  });
```

Comment:
These are good tests but should be split up into different tests so its easier to read